src/ert/shared/feature_toggling.py

```diff
@@ -1,71 +1,132 @@
 import logging
+import os
+import re
 from argparse import ArgumentParser
 from copy import deepcopy
-from typing import Optional
+from typing import Dict, Optional, Union
 
 from ert.namespace import Namespace
 
 
 class _Feature:
-    def __init__(self, default_enabled: bool, msg: Optional[str] = None) -> None:
-        self.is_enabled = default_enabled
+    def __init__(
+        self, default: Optional[bool], msg: Optional[str] = None, optional: bool = False
+    ) -> None:
+        self._value = default
+        self.env_var: Optional[bool] = default
         self.msg = msg
+        self.optional = optional
+
+    def validate_value(self, value: Union[bool, str, None]) -> Optional[bool]:
+        if type(value) is bool or value is None:
+            return value
+        elif value.lower() in ["true", "1"]:
+            return True
+        elif value.lower() in ["false", "0"]:
+            return False
+        elif self.optional and value.lower() in ["default", ""]:
+            return None
+        else:
+            raise ValueError(
+                f"This option can only be set to {'True/1, False/0 or Default/<empty>' if self.optional else 'True/1 or False/0'}"
+            )
+
+    @property
+    def value(self) -> Optional[bool]:
+        return self._value
+
+    @value.setter
+    def value(self, value: Optional[bool]) -> None:
+        self._value = self.validate_value(value)
 
 
 class FeatureToggling:
-    _conf_original = {
+    _conf_original: Dict[str, _Feature] = {
         "new-storage": _Feature(
-            default_enabled=False,
+            default=False,
             msg=(
                 "The new storage solution is experimental! "
                 "Thank you for testing our new features."
             ),
         ),
         "scheduler": _Feature(
-            default_enabled=False,
-            msg="Use Scheduler instead of JobQueue",
+            default=None,
+            msg="Force enable/disable the usage of the new scheduler.",
+            optional=True,
+        ),
+        "old-storage": _Feature(
```

Comment:
There is no such thing as `old-storage` and `new-storage` has not been doing anything since b0ef3f5f6d8d92af79b3cc5e3769a93ab68c52a3 . You can remove this.

We can maybe make `FeatureToggling` not be a singleton for testing purposes. Eg:
```py
class FeatureToggling:
    def __init__(self, config: Mapping[str, _Feature]) -> None:
        self._config

    # [..]

FEATURE_TOGGLING = FeatureToggling({
    "scheduler": _Feature(...)
})

def is_enabled(feature: str) -> bool:
    return FEATURE_TOGGLING.is_enabled(feature)
```

Everywhere uses `is_enabled` function directly while feature toggling tests can test that eg. hyphenated names work correctly by creating a new instance of `FeatureToggling`.

Reply:
old-storage was something I was testing with. came back when I messed up a rebase/squash. removed new-storage and all code referencing it now.
Changing FeatureToggling to not be a singleton can be a different PR/issue